frontend/src/metabase/entities/fields.js

```diff
@@ -88,11 +88,18 @@ const Fields = createEntity({
     }),
 
     updateField(field, opts) {
-      return Fields.actions.update(
-        { id: field.id },
-        field,
-        notify(opts, field.display_name, t`updated`),
-      );
+      return async dispatch => {
+        const result = await dispatch(
+          Fields.actions.update(
+            { id: field.id },
+            field,
+            notify(opts, field.display_name, t`updated`),
+          ),
+        );
+        // Field values needs to be fetched again once the field is updated metabase#16322
+        await dispatch(Fields.actions.fetchFieldValues(field));
```

Comment:
@kulyk I changed it so that it uses the same structure for both actions.

`dispatch(thunkAction)` rather than `thunkAction(dispatch, getState)`.